app/models/fe/element.rb

```diff
@@ -192,6 +195,18 @@ def update_any_previous_conditional_elements
       end
     end
 
+    def update_page_all_element_ids
+      pages.each do |p| p.rebuild_all_element_ids end
+
+      if question_grid
+        question_grid.update_page_all_element_ids
+      elsif question_grid_with_total
+        question_grid_with_total.update_page_all_element_ids
+      elsif choice_field
+        choice_field.update_page_all_element_ids
+      end
```

Comment:
I like that better too.  I think I also need to take out choice_field since we don't want options in choice_fields to count as questions (both for counting questions and as questions that require answers)